Examples/HDL/Digital_Filtering/HDL_DUC/README.md

```diff
@@ -0,0 +1,93 @@
+# Digital Upconversion (DUC)
+
+## Introduction
+
+This example demonstrates a Digital Upconversion (DUC) algorithm implemented on Programmable Logic using AMD IP blocks.
+
+See also the [DUC implemented on AI Engine devices](https://github.com/Xilinx/Vitis_Model_Composer/tree/2024.2/Examples/AIENGINE/DSPlib/DUC).
+
+## Algorithm
+
+The DUC design consists of multi-stage finite impulse rate (FIR) filters, a direct digital synthesizer (DDS) and a mixer. 
+
+![](./Images/duc_block_diagram.png) 
+
+The DUC specification is as follows:
+* 4 stages of FIR filter, with interpolation ratio of 2 in each stage and an overall interpolation ratio of 16.
+* The first stage is a 64 tap square raised root cosine (SRRC) filter, and the next three stage filters are half-band (HB) interpolate by 2 FIR filters. 
+
+## Example Model
+
+The example model compares a Simulink implementation of the DUC with an HDL implementation.
+
+![](./Images/model.png) 
+
+The input to both designs is white noise with a sample rate of 50 MSPS.
+
+![](./Images/input_signal.png) 
+
+### Simulink Reference Design
+
+In the reference design, the 4 filter stages are implemented with Simulink's **FIR Interpolation** block.
+
+![](./Images/simulink_reference.png) 
+
+The **Gain** blocks between each stage control bit growth and ensure that the input to each filter is 32 bits.
+
+### PL DUC Design
+
+The filters in the PL design are implemented using the FIR Compiler block from the **HDL/DSP/AXI-S** library.
+
+![](./Images/pl_model.png) 
+
+The **Shift** blocks between each stage control bit growth and ensure that the input to each filter is 17 bits.
+
+The mixer is implemented using:
+
+* DDS Compiler block to implement a complex sinusoid.
+* DSP58 blocks to implement a complex multiply operation. 4 DSP58s are arranged in cascade to achieve high performance and close timing at 800 MHz.
```

Comment:
I recommned to add a sentence on DSP58. That it is a new IP in Versal devices and what the advantage is over DSP48.